### PR TITLE
[FIXED JENKINS-37232] - starting with 2.7 the ATH began to fail during create item

### PR DESCRIPTION
**Pull request by kzantow**
[JENKINS-37232](http://issues.jenkins-ci.org/browse/JENKINS-37232)

Firefox has an issue sending blur events when running the ATH. This simulates the blur event for a specific field that needs it during item creation.

@reviewbybees

### CODE REVIEW

**Comment by varmenise** [comment]
🐝

**Comment by amuniz** [comment]
mega-:bee:

**Comment by recena** [comment]
Waiting for the results from the build....

File: src/main/java/org/jenkinsci/test/acceptance/po/JobsMixIn.java [hunk 1]
```diff
@@ -22,6 +22,21 @@ public JobsMixIn(ContainerPageObject context) {
...
+        // since 2.7, a bug in Firefox webdriver may prevent the blur event from triggering
+        // properly, so we manually execute a blur event here, see: JENKINS-37232
+        try {
+            executeScript(
+                "var obj = document.getElementById('name');"
+                + "var ev = document.createEvent('MouseEvents');"
+                + "ev.initEvent('blur', true, false);"
+                + "obj.dispatchEvent(ev);"
+                + "return true;");
+        } catch(Exception e) {
```
Comment (recena) [inline src/main/java/org/jenkinsci/test/acceptance/po/JobsMixIn.java]: `catch (Exception e) {`

**Comment by recena** [comment]
@kzantow Do you have any reference (link) about this bug in the Firefox webdriver?

**Comment by kzantow** [comment]
@recena fixed the formatting and added a link to (one of) the issues reported on this: https://github.com/seleniumhq/selenium-google-code-issue-archive/issues/7346

**Comment by amuniz** [comment]
re-:bee:

**Comment by oleg-nenashev** [comment]
🐝 
@samatdav What will happen when your PR gets merged?

**Comment by kzantow** [comment]
@oleg-nenashev I assume https://github.com/jenkinsci/jenkins/pull/2388 is what you're referring to? There might need to be more changes here, this blur change will probably still be needed if the existing code is still being used to validate the name field.

**Comment by samatdav** [comment]
@kzantow I guess Oleg may refer to https://github.com/jenkinsci/jenkins/pull/2416
@oleg-nenashev I am not sure but it seems like my PR should be fine. Since I do not change the standard behavior of the `blur` event. I only add another event.

**Comment by recena** [comment]
👍

**Comment by olivergondza** [comment]
I wonder what kind of FF problem that is. I experienced problems with events not sent / handled differently when firefox did not have focus. It works as expected in VNC, though. Is this one of those cases?

In that case I would really prefer requiring browser to have focus to run ATH (using dedicated display would be a requirement) over ad-hoc javascript fixups to compensate what is a problem between selenium and target browser (as I understand it).

**Comment by kzantow** [comment]
@olivergondza it would be nice to have consistent behavior for Firefox, but really we just need this to work and people won't necessarily have access to run this through some VNC environment without a lot of setup. It would be irritating to add tests if you can't just run an individual one on your local machine, which wouldn't bode well for actually having tests added. And hopefully we can all agree having more tests is a good thing.

**Comment by kzantow** [comment]
@reviewbybees done

**Comment by kzantow** [comment]
@reviewbybees again :)

**Comment by mafraba** [comment]
:bee:

**Comment by recena** [comment]
👍

**Pull request merged**